matrix-sdk/src/main/java/org/matrix/androidsdk/data/timeline/TimelineInvitedRoomSyncHandler.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2018 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.matrix.androidsdk.data.timeline;
+
+import android.support.annotation.NonNull;
+
+import org.matrix.androidsdk.data.Room;
+import org.matrix.androidsdk.rest.model.Event;
+import org.matrix.androidsdk.rest.model.sync.InvitedRoomSync;
+
+import javax.annotation.Nullable;
+
+/**
+ * This class is responsible for handling the invitation room events from the SyncResponse
+ */
+class TimelineInvitedRoomSyncHandler {
+
+    private final Room mRoom;
+    private final TimelineLiveEventHandler mLiveEventHandler;
+    private final InvitedRoomSync mInvitedRoomSync;
+
+    TimelineInvitedRoomSyncHandler(@NonNull final Room room,
+                                   @NonNull final TimelineLiveEventHandler liveEventHandler,
+                                   @Nullable final InvitedRoomSync invitedRoomSync) {
+        mRoom = room;
+        mLiveEventHandler = liveEventHandler;
+        mInvitedRoomSync = invitedRoomSync;
+    }
+
+    /**
+     * Handle the invitation room events
+     */
+    public void handle() {
```

Comment:
It's strange that `InvitedRoomSync` is not passed as a param here, but in the constructor.

Reply:
It's a common pattern for breaking dependencies. So, it's just the first step for introducing tests and then refactor !

Reply:
ok